### ISSUE TITLE
Steam3 timeout when using latest version

### ISSUE THREAD

**Issue by phamleduy04** [SteamRE/DepotDownloader#210]
I tried to use the same command with same login creds in v2.3.6 and its working but not the latest version (v2.4.1)

**Comment by azuisleet** [comment]
Please run with -debug and provide the log.

**Comment by Aida-Enna** [comment]
It appears? to be an issue with Steam.
![image](https://user-images.githubusercontent.com/5483362/119889888-2287d100-bf05-11eb-9cfc-9852585d28e1.png)

**Comment by phamleduy04** [comment]
but the old version still work

**Comment by phamleduy04** [comment]
```
>dotnet .\DepotDownloader.dll -app 620980 -depot 620981 -manifest 7624554893344753887 -username MYUSERNAME -password MYPASSWORD -debug


Connecting to Steam3...[ServerList] Resolving server list
[ServerList] Server list provider had no entries, will query SteamDirectory
Timeout connecting to Steam3.
[352593ad2e04410fb626df1ae278c384/CMClient] Connection cancelled before a server could be chosen.
Disconnected from Steam
Unable to get steam3 credentials.
Error: InitializeSteam failed
```

**Comment by azuisleet** [comment]
Do you have a firewall that is preventing the application from connecting to api.steampowered.com?

**Comment by phamleduy04** [comment]
I don't think so bc I'm using old version and its works fine

**Comment by azuisleet** [comment]
Between the two releases of SteamKit https://github.com/SteamRE/SteamKit/compare/2.3.0...2.4.0-Alpha.2, I don't see anything that would cause a regression in SteamDirectory or WebAPI.

However, the default timeout on WebAPI is 100 seconds, which is much longer than I would expect.

The fact that it's taking 100 seconds to time out suggests some other network issue that's causing problems.

**Comment by yaakov-h** [comment]
What .NET Runtime, operating system, etc. are you running on?

If the old version still works I suspect it may not be network-layer-related.

**Comment by azuisleet** [comment]
2.4.2 enables additional diagnostics for HTTP when -debug is specified. Give it a shot and provide a log.

https://github.com/SteamRE/DepotDownloader/releases/tag/DepotDownloader_2.4.2

**Comment by azuisleet** [comment]
This should be resolved in 2.4.3 by disabling IPv6 on HTTP sockets. If you still have this issue, feel free to re-open it.

https://github.com/SteamRE/DepotDownloader/releases/tag/DepotDownloader_2.4.3
